### PR TITLE
fixed empty response on get_block and get_block_by_hash

### PR DESCRIPTION
**Pull request by dabasov**
the problem was that events were not written in the events db, now we store all blocks that were added to the chain

### CODE REVIEW

**Review by Sriep — APPROVED** [review]

File: code/go/0chain.net/chaincore/chain/entity.go [hunk 1]
```diff
@@ -693,6 +693,9 @@ func (c *Chain) addBlock(b *block.Block) *block.Block {
 		return eb
 	}
 	c.blocks[b.Hash] = b
+
+	c.pushBlockEvent(b)
```
Comment (peterlimg) [inline code/go/0chain.net/chaincore/chain/entity.go]: We only save finalized blocks to event db, so perhaps do it in https://github.com/0chain/0chain/blob/9b72325418072680429bc00b654367b1234b6af9/code/go/0chain.net/chaincore/chain/protocol_block.go#L351.

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/chaincore/block/block_eventdb.go [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package block
 
 import (
+	"0chain.net/core/util"
```
Comment (peterlimg) [inline code/go/0chain.net/chaincore/block/block_eventdb.go]: Run goimports to format

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/chaincore/chain/entity.go (hunk 1 above)
Comment (dabasov) [reply code/go/0chain.net/chaincore/chain/entity.go]: I've done it intentionally, we might need all proposed blocks here, not only finalized blocks, at least all notarized so I suggest to save them here

Comment (peterlimg) [reply code/go/0chain.net/chaincore/chain/entity.go]: okay, we can leave this and consider to do finalized blocks only if we find those none finalized proposed blocks are not being used at all in the future.

**Review by peterlimg — APPROVED** [review]
lgtm

File: code/go/0chain.net/chaincore/chain/entity.go (hunk 1 above)
Comment (peterlimg) [inline code/go/0chain.net/chaincore/chain/entity.go]: Oh, event db only exists in sharders, so we can not do this for miners.

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/chaincore/chain/entity.go (hunk 1 above)
Comment (dabasov) [reply code/go/0chain.net/chaincore/chain/entity.go]: oooops

**Review by peterlimg — APPROVED** [review]
lgtm

**Pull request merged**